services/slam/server_test.go

```diff
@@ -149,19 +305,71 @@ func TestServer(t *testing.T) {
 		test.That(t, resp, test.ShouldBeNil)
 	})
 
+	t.Run("failing get pointcloud map stream function", func(t *testing.T) {
+		// GetPointCloudMapStreamFunc failure
+		injectSvc.GetPointCloudMapStreamFunc = func(ctx context.Context, name string) (func() ([]byte, error), error) {
+			f := func() ([]byte, error) {
+				return []byte{}, io.EOF
+			}
+			return f, errors.New("failure to get pointcloud map")
```

Comment:
I recommend making f `nil` in this case given that the callback function caller should never invoke the callback function given an error will always be returned.